zio/src/main/scala/xs4s/ziocompat/ZIOSyntax.scala

```diff
@@ -0,0 +1,41 @@
+package xs4s.ziocompat
+
+import javax.xml.stream.XMLEventReader
+import javax.xml.stream.events.XMLEvent
+import xs4s.XmlElementExtractor
+import xs4s.generic.Scanner
+import xs4s.syntax.core._
+import zio.stream.{Stream, ZStream}
+import zio.Runtime.Managed
+import zio.ZManaged
+import java.io.IOException
+import zio.blocking.Blocking
+
+/**
+  * Utilities to enhance xs4s interaction with ZIO
+  */
+trait ZIOSyntax {
+  implicit class RichScanner[I, S, O](scanner: Scanner[I, S, O]) {
+    /** Create an ZIO-FS2-style Pipe from the scanner */
+    def zioPipe[R]: Pipe[R, I, O] =
+      _.mapAccum(scanner.initial)((s: S, i: I) => (scanner.scan(s, i), scanner.collect(s))).collectSome
```

Comment:
might be good to expand symbol names eg 'scannerState', 'input', etc, just for readability as often people will evaluate while looking at GitHub only (in all the files)